File: src/main/java/com/karmanno/plugins/tasks/PrintVersionTask.java
package com.karmanno.plugins.tasks;

import com.karmanno.plugins.services.PrepareVersionService;
import com.karmanno.plugins.utils.GitUtils;
import org.eclipse.jgit.api.Git;
import org.gradle.api.DefaultTask;
import org.gradle.api.Project;
import org.gradle.api.tasks.TaskAction;

public class PrintVersionTask extends DefaultTask {

    @TaskAction
    public void doTask() {
        Project project = getProject();
        Git git = GitUtils.gitRepo(project);
        String version = new PrepareVersionService().getVersion(git).printVersion();
        System.out.println(version);
    }
}

File: src/main/kotlin/com/karmanno/plugins/task/PrintVersionTask.kt
package com.karmanno.plugins.task

import org.gradle.api.DefaultTask
import org.gradle.api.tasks.Input
import org.gradle.api.tasks.Optional
import org.gradle.api.tasks.TaskAction
import org.gradle.api.tasks.options.Option

open class PrintVersionTask: DefaultTask() {
    @Input
    @Optional
    @Option(description = "Release branch, if it is different to master", option = "releaseBranch")
    var releaseBranch: String? = null

    @TaskAction
    fun doTask() {
        println(GenericIncreaseVersionTask().calculateVersionString(project, releaseBranch).second)
    }
}